packages/dx-react-grid/src/plugins/table-inline-cell-editing.tsx

```diff
@@ -0,0 +1,154 @@
+                rowChanges,
+                isColumnEditingEnabled,
+            }, {
+              changeRow,
+              startEditCells,
+              stopEditCells,
+              commitChangedRows,
+              cancelChangedRows,
+            }) => {
+              const { tableRow, tableColumn } = params;
+              const { rowId, row } = tableRow;
+              const { column } = tableColumn;
+              const { name: columnName } = column!;
+
+              if (isInlineEditTableCell(tableRow, tableColumn, editingCells)) {
+                const changedRow = {
+                  ...row, ...getRowChange(rowChanges, rowId!),
+                };
+
+                const value = getCellValue(changedRow, columnName);
+                const onValueChange = (newValue: any) => {
+                  const changeArgs = {
+                    rowId,
+                    change: createRowChange(changedRow, newValue, columnName),
+                  };
+                  changeRow(changeArgs);
+                };
+                const onKeyDown = (key) => {
+                  if (key === 'Enter') {
+                    commitChangedRows({ rowIds: [rowId] });
+                    stopEditCells({ editingCells: [{ rowId, columnName }] });
+                  } else if (key === 'Escape') {
+                    cancelChangedRows({ rowIds: [rowId] });
+                    stopEditCells({ editingCells: [{ rowId, columnName }] });
+                  }
+                };
+                const onBlur = () => {
+                  commitChangedRows({ rowIds: [rowId] });
+                  stopEditCells({ editingCells: [{ rowId, columnName }] });
+                };
+                const onFocus = selectTextOnEditStart ? e => e.target.select() : () => {};
+                const editingEnabled = isColumnEditingEnabled(columnName);
+
+                return(
+                  <TemplatePlaceholder
+                    name="valueEditor"
+                    params={{
+                      column,
+                      row,
+                      value,
+                      onValueChange,
+                      disabled: !editingEnabled,
+                    }}
+                  >
+                    {content => (
+                      <EditCell
+                        {...params}
+                        row={row}
+                        column={column!}
+                        value={value}
+                        editingEnabled={editingEnabled}
+                        onValueChange={onValueChange}
+                        autoFocus
+                        onKeyDown={({ key }) => onKeyDown(key)}
+                        onBlur={onBlur}
+                        onFocus={onFocus}
+                      >
+                        {content}
+                      </EditCell>
+                    )}
+                  </TemplatePlaceholder>
+                );
+              }
+
+              if (startEditAction === 'click') {
```

Comment:
Can we collapse these two renders into one? I mean pass onClick or onDblClick by condition only.